src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests.cs

```diff
@@ -799,6 +799,67 @@ class C
             Assert.Same(parseOptions, passedOptions);
         }
 
+        [Fact]
+        public void AdditionalFiles_Are_Passed_To_Generator()
+        {
+            var source = @"
+class C 
+{
+}
+";
+            var parseOptions = TestOptions.Regular;
+            Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+            compilation.VerifyDiagnostics();
+            Assert.Single(compilation.SyntaxTrees);
+
+            var texts = ImmutableArray.Create<AdditionalText>(new InMemoryAdditionalText("a", "abc"), new InMemoryAdditionalText("b", "def"));
+
+            ImmutableArray<AdditionalText> passedIn = default;
+            var testGenerator = new CallbackGenerator(
+                onInit: (i) => { },
+                onExecute: (e) => passedIn = e.AdditionalFiles
+                );
+
+            GeneratorDriver driver = CSharpGeneratorDriver.Create(new[] { testGenerator }, parseOptions: parseOptions, additionalTexts: texts);
+            driver.RunGeneratorsAndUpdateCompilation(compilation, out _, out _);
+
+            Assert.Equal(2, passedIn.Length);
+            AssertEx.SetEqual(texts, passedIn);
```

Comment:
This makes it seem like the order the generator receives the additional files is not important. It feels like the order probably is important for determinism, though. If that is the case, consider replacing these two asserts with just `Assert.Equal(texts, passedIn)`.

Reply:
Ah yeah, its because the actual enumerable is different, but the elements are not. I'll make it deterministic.